### ISSUE TITLE
2/4 HF4 CTD AI turn

### ISSUE THREAD

**Issue by calloatti** [LoneGazebo/Community-Patch-DLL#7486]
_1. Mod version (i.e Date - 12/1):_

2/4 HF4

_2. Mod list (if using Vox Populi only, leave blank):_

Extra mods are in use. The MODS and cache folders are included in full the zip. Not all mods in the MODS folder are enabled, that is why I am including the cache folder.

_3. Error description:_ 

CTD in turn 480 after clicking NEXT TURN and Babylon starts its turn

_4. Steps to reproduce (optional):_

Use the folders included in the zip file to setup CIV5 modded instance, load saved game CTD, click NEXT TURN

---------------------------
Supporting information:

zip file includes logs folder, and cvminidump only:

[logs.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/5956464/logs.zip)

full zip with everything included to repro:

https://drive.google.com/file/d/1tdq724ZOY436Si5wgRegG30lVTPsr4og/view?usp=sharing

**Comment by RecursiveVision** [comment]
It's always difficult to debug when lots of extra mods are in use. The cache folder is not very useful, it would be better if you included the list of enabled mods instead.

**Comment by calloatti** [comment]
I think this will help then:

```
ModID	Version	name	Enabled
d1b6328c-ff44-4b0d-aad7-c657f83610cd	94	(1) Community Patch	1
8411a7a8-dad3-4622-a18e-fcc18324c799	13	(2) Community Balance Overhaul	1
eead0050-1e3f-4178-a91f-26cf1881ac39	28	(3) City-State Diplomacy Mod for CBP	1
c7bf7064-d1b1-4708-9e93-7a1560868582	11	(4) C4DF - CBP	1
be2d13ee-d18d-410e-8f04-7cd19cd99a98	155	(5) More Luxuries - CBO Edition (5-14b)	1
24923240-e4fb-4bf6-8f0e-6e5b6cf4d3c2	1	(6a) Community Balance Overhaul - Compatibility Files (EUI)	1
ef8387f0-b053-4c84-90ce-c3c5cb21a580	5	(7) Community Events	1
0f1eab20-b84e-4277-a3cc-f772b952be54	3	(7a) Promotion Icons for VP	1
1f0a153b-26ae-4496-a2c0-a106d9b43c95	22	(7b) UI - Promotion Tree for VP	1
d7d2f630-cf19-4663-a5a3-4ed417b96af7	4	(game) Capture+++	1
9e735b12-3815-4ce3-9845-5396afb189da	11	(ui) Trade Opportunities for VP	1
d4a7e692-5d05-4436-bc84-13236d75575f	8	(ui) Wonder Planner for VP	1
2dec3dd9-f21f-4659-afdc-dd64085cd01d	2	Barbarians - Unlimited Exp	1
292510fb-21c5-45fe-9fb5-fc5e62e67e5d	1	Buildings - Watermill Near River	1
31df5e77-6a9e-4312-8e74-980fffdd5848	1	Buildings - Windmill Near Flat	1
a85815ab-f3c1-4daf-9700-6199975f6d79	1	Caravan Range Extension	1
f1830fcf-5e1b-4acb-aed7-c923e1f49562	1	Diplomacy - Auto-Denounce	1
fe4937f4-eb6e-4197-b0ff-2c53cc1b34e6	13	Eras - Atomic Age	1
43e7cd1c-dea2-4e60-b6dc-9cffc2c3fcfe	7	FlagPromotions	1
7bab5397-0ba6-4eb0-9af4-dba46f920258	1	Global - City Jungle Bonus	1
72e3495c-054e-4a15-9c49-70e2ca197621	7	Global - City Working Distance	1
290fa859-b8d1-4d35-ba71-5deae61f0da7	2	Global - Civilians Move After Purchase	1
f48e14fc-ac0b-429e-a142-ba1ccbf4c7e2	1	Global - Jungle Production	1
a526e077-5066-4268-a23f-95a562e709e1	2	Global - Local Generals	1
eba7d61d-b6ea-428c-bd36-c5fcb2f68981	1	Global - National Wonders  Exclude Razing	1
d6dd2d43-f73d-4608-82f2-881898413eb3	1	Global - No Ocean Plundering	1
34d9746b-2f1e-4ac1-ab24-6f1ad4a6acbd	1	Global - Religious Settlers	1
48d98c35-8396-44c2-b45b-ed48cf6d3eff	2	Goody Huts - No Auto-Mapping	1
0a2d1a47-cd9d-47a0-b3ed-9fc36c4edb5b	7	Goody Huts - Tech Refund	1
18d7c770-a13a-4563-9e52-cb81b7d75537	1	Improvement - Farm Replacement	1
aec5d10d-f00f-4fc7-b330-c3a1e86c91c3	22	InfoAddict	1
170c8ed1-b516-4fe2-b571-befeac39d220	39	Ingame Editor	1
2a904f73-4d28-4d2a-ac57-a1ac3cc42ddc	1	Late Game Range BNW	1
23fef741-4cf5-41c4-9e34-9d350494fb52	1	Meaningful City State Colors	1
98c24719-d2a2-4dd0-a969-4e53cc86c763	3	Monopoly Buildings for Vox Populi	1
e9ead585-4f3b-4087-bcfa-9dc21453fe0b	1	PerfectWorld3	1
99506232-2c32-4800-8beb-305f3240f169	2	Promotions - All Siege Units Indirect Fire	1
3885caca-b878-4d3a-8794-9fcde08f80bd	5	Promotions - Extra Cargo	1
0410e638-98da-4d02-930a-daf4c3a120ef	10	Quick Turns	1
34feb829-33fb-4241-956f-462e6877e070	15	Really Advanced Setup	1
5114ca56-5621-43cb-a024-e0884b6045da	9	Reforestation	1
b4e8428d-3eae-4511-a8de-daa268fdbf22	1	Religion - Bonus Beliefs	1
0fe65d15-fc91-4a8d-b3a7-a33090c13008	15	Religion - Improved Customization	1
960e4924-8e48-4e12-b123-49c6b605bc79	1	Religion - No AI Free Pottery	1
cb40b6f3-ebee-468d-902d-6f5a5ffc59f3	3	Resource - Granary Includes Bison	1
3ccda7fd-a229-4e42-a9cb-e2ea9a39e243	1	Resources - Copper buff	1
72faea6d-0c72-4f10-bf0d-c6240f11afa0	2	Terrain - Poor Tiles Tweak	1
d77f9a36-6186-470d-aad3-cb78426f004d	15	UI - City Production	1
eab3f33b-c711-49d5-a1c1-98e47ceb190c	4	UI - Destination	1
f30c5651-8fce-44c8-abe6-4bc9bd582098	1	UI - Filters in Trade Route Overview (Vox Populi)	1
868e845c-a185-4e00-b0f9-f0c971e0ec1c	2	UI - Overlay Antiquities	1
a685c6da-bc8f-445f-9e83-5d606b136887	8	UI - Overlay Barbarians	1
ef3f23d4-a288-4c50-9937-0934dd2ed07e	3	UI - Overlay Luxuries	1
f2968890-fe71-41d4-b609-805097365935	11	UI - Overlay Resources	1
e915c6ee-16bc-44cf-9a2f-15746f414c63	7	UI - Overlay Wonders	1
4ad8de1c-a52a-4752-8e61-e73edcecf8dd	13	UI - Religion Spread	1
dab4ce8a-73a5-4beb-81c1-808e4baf45a8	4	UI - Small Resource Icons	1
206490e3-31ed-493a-a2fb-af645a42bed8	3	Units - Scouts And Explorers Ignore Borders	1
00a28291-c503-4edf-8d76-0da4f86669d2	1	Units - Starting Scout	1
3c90cf1d-7ff4-4277-a1bf-bec4c431fdfc	3	Units - Subs Ignore Borders	1
6334f1c2-d904-4f57-8fed-787fd1712a6b	2	Units - Subs Immune Under Ice	1
75d92b9d-b908-45fe-b492-5955eab82b78	3	War - Less Damaged Captured Cities	1
726cd01f-8ea2-4438-8085-21ee14bd9500	7	Wonder Race	1
ccf47da8-4d56-4c74-a416-ab07a98484a8	2	Worker Mountaineering	1
```

I used this to get it:

```
select modproperties.ModID, modproperties.version, modproperties.value as 'name', 
mods.enabled from modproperties inner join Mods on modproperties.ModID = mods.ModID 
where modproperties.name = 'Name' and mods.Enabled = 1 order by modproperties.Value
```

**Comment by calloatti** [comment]
One thing I tried was enabling "Quick Combat" when at war with an AI player in the "Quick Turns" mod. That got past the CTD in this case, but got more CTD later no matter the setting. I also tried disabling the "Quick Turns" mod and it still crashes.

**Comment by ilteroi** [comment]
nothing of interest in the crash dump

**Comment by RecursiveVision** [comment]
@calloatti I'll see if I can debug it, but no promises with all those mods.

You can remove "Diplomacy - Auto Denounce" by the way, I removed that mod's function from VP. You can now denounce while at war so it's obsolete.

**Comment by RecursiveVision** [comment]
For whatever reason, your Community Patch folder refuses to show up in my mod list and I can't activate it.

**Comment by RecursiveVision** [comment]
I also suggest deleting the mods you don't use from the folder, especially that DLL mod...

**Comment by calloatti** [comment]
Yes I noticed the same thing about the mod not showing up in the mods list. A workaround I found is renaming the folder while CIV is open in the mods list, and then renaming it back. Then it shows up in the mods list.

Will delete unused mods and see what happens, thank you.

**Comment by calloatti** [comment]
I am in the process of deleting and unsubscribing all the inactive mods, will report back what happens. Steam acting up as usual: "Failed loading profile data, please try again later." No way to unsubscribe to mods. I guess this will take a while.

**Comment by calloatti** [comment]
Still crashing, closing this one and will open a new issue, since I updated to 2/4 hotfix 5 and deleted disabled mods.

**Comment by calloatti** [comment]
Not going to open a new issue, managed to avoid the CTD by enabling Quick Combat for a few turns until I wiped off Babylon, then I could continue playing with no more crashes.

**Comment by calloatti** [comment]
I hope no one minds, I am going to just rant here. The Steam Workshop system and the way the game handles the mods you are subscribed to is all a big pile of s***t.

Something as simple as deleting the cache and mods folder and making civ5 redownload the mods I am subscribed to is taking me hours and multiple restarts.

I am at the stage where I managed to get all but 2 mods installed. One I just quit and unsubscribed, the other one, that I can see in the list of mods I am subscribed to in the workshop, will not download/install no matter how many times I try.

Then this, for that mod:

![image](https://user-images.githubusercontent.com/12817862/107612775-c1721100-6c25-11eb-979f-6c2e113c97b4.png)

Steam, I just clicked in the link to the mod, that you listed as one of the mods I am subscribed to...

![image](https://user-images.githubusercontent.com/12817862/107612841-e5cded80-6c25-11eb-8a80-99597fea7f04.png)
